### DISCUSSION TITLE
WHy do I get the loader loader data when I hit a route directly but the correct HTML when I navigate there

### DISCUSSION THREAD

**Discussion by dagda1** [remix-run/remix#7156] [Q&A]
I have a remix site which uses [aws architect](https://arc.codes/docs/en/get-started/quickstart) problem that I don't understand.

This only happens when when deployed to AWS.

I have the following route `posts._index.tsx`

![route](https://github.com/remix-run/remix/assets/118328/1b751e46-b67f-426e-acc2-e627e668f876)

```ts
export const loader: LoaderFunction = async () => {
  return getPosts();
};

export default function PostsHome() {
  const posts = useLoaderData<PostData[]>();

  return <Posts posts={posts} />;
}
```

If I go directly to the [route](https://staging.frontendrescue.com/posts), it return `application/json` which is the JSON returned from the loader function.

![json](https://github.com/remix-run/remix/assets/118328/d10c48b3-6e94-4de9-84fd-589c13762514)

But If I navigate there by clicking the `blog` link on the [home page](https://staging.frontendrescue.com), it returns `text/html`.

![json](https://github.com/remix-run/remix/assets/118328/eb6ea5d9-ee8b-45e4-b41e-f1e037f9560b)

Am I doing something wrong?

**Comment by b-goldney** [comment]
Did you ever figure out a solution to this? I'm facing the same issue.

EDIT:
I just found your blog with the solution [here](https://frontendrescue.com/posts/2023-08-19-remix-cloudfront). Thank you for resolving. I've been stuck on this issue for way too long now.

**Reply by dagda1** [reply]
@b-goldney really glad it helped someone else.  I was tearing my hair out over that one.

**Comment by dagda1** [comment]
> Did you ever figure out a solution to this? I'm facing the same issue.
> 
> EDIT: I just found your blog with the solution [here](https://frontendrescue.com/posts/2023-08-19-remix-cloudfront). Thank you for resolving. I've been stuck on this issue for way too long now.
